lib/build.js

```diff
@@ -180,22 +180,53 @@ export default {
 
       this.child.on('close', (exitCode) => {
         this.child = null;
-        this.errorMatcher.set(target.errorMatch, cwd, stdout + stderr);
+        this.errorMatcher.set(target, cwd, stdout + stderr);
 
         let success = (0 === exitCode);
         if (atom.config.get('build.matchedErrorFailsBuild')) {
-          success = success && !this.errorMatcher.hasMatch();
+          success = success && !this.errorMatcher.getMatches().some(match => match.criticality === 'error' || match.type && match.type.toLowerCase() === 'error');
+        }
+
+        function extractRange(json) {
+          return [
+            [ (json.line || 1) - 1, (json.col || 1) - 1 ],
+            [ (json.line_end || json.line || 1) - 1, (json.col_end || json.col || 1) - 1 ]
+          ];
+        }
+        function normalizePath(p) {
+          return require('path').isAbsolute(p) ? p : require('path').join(cwd, p);
+        }
+        function typeToSeverity(type) {
+          switch (type && type.toLowerCase()) {
+            case 'error': return 'error';
+            case 'warning': return 'warning';
+            case 'trace': return 'info';
+            case 'note': return 'info';
+            default: return null;
+          }
+        }
+        function cleanSeverity(severity) {
+          switch (severity && severity.toLowerCase()) {
+            case 'error': return 'error';
+            case 'warning': return 'warning';
+            case 'info': return 'info';
+            default: return null;
+          }
         }
 
-        const npath = require('path');
         this.linter && this.linter.setMessages(this.errorMatcher.getMatches().map(match => ({
-          type: 'Error',
+          type: match.type || 'Error',
           text: match.message || 'Error from build',
-          filePath: npath.isAbsolute(match.file) ? match.file : npath.join(cwd, match.file),
-          range: [
-            [ (match.line || 1) - 1, (match.col || 1) - 1 ],
-            [ (match.line_end || match.line || 1) - 1, (match.col_end || match.col || 1) - 1 ]
-          ]
+          filePath: normalizePath(match.file),
+          severity: cleanSeverity(match.severity) || typeToSeverity(match.type) || 'error',
```

Comment:
I feel this may be a bit too much. Why do we need to introduce both `type` and `severity` ?